rust/protocol/src/session_cipher.rs

```diff
@@ -595,23 +574,13 @@ fn decrypt_message_with_state<R: Rng + CryptoRng>(
         ));
     }
 
-    let ptext = match crypto::aes_256_cbc_decrypt(
+    let ptext = match crypto::cbc::aes_256_cbc_decrypt(
         ciphertext.body(),
         message_keys.cipher_key(),
         message_keys.iv(),
     ) {
         Ok(ptext) => ptext,
-        Err(crypto::DecryptionError::BadKeyOrIv) => {
-            log::warn!(
-                "{} session state corrupt for {}",
-                current_or_previous,
-                remote_address,
-            );
-            return Err(SignalProtocolError::InvalidSessionStructure(
-                "invalid receiver chain message keys",
-            ));
-        }
```

Comment:
This is the change that convinced me this is a good idea: at first I thought "what happens if those fields are corrupt"? But they can't be, because we already check that when getting the message keys out of the session state, and that wasn't in the type system anywhere.